features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -87,21 +87,24 @@ class Base64Serializer private constructor(
     }
 
     @WorkerThread
-    private fun convertBmpToBase64(bitmap: Bitmap): String {
-        val byteArrayOutputStream = webPImageCompression.compressBitmapToStream(bitmap)
+    private fun convertBmpToBase64(initialBitmap: Bitmap): String {
+        var base64Result = ""
 
-        if (isOverSizeLimit(byteArrayOutputStream.size())) {
-            return ""
-        }
+        val scaledBitmap = drawableUtils.downscaleBitmapToApproxSize(initialBitmap)
 
-        val base64: String
-        try {
-            base64 = base64Utils.serializeToBase64String(byteArrayOutputStream)
-        } finally {
-            bitmap.recycle()
+        scaledBitmap?.let {
+            val byteArrayOutputStream = webPImageCompression.compressBitmapToStream(it)
+
+            try {
+                if (!isOverSizeLimit(byteArrayOutputStream.size())) {
+                    base64Result = base64Utils.serializeToBase64String(byteArrayOutputStream)
```

Comment:
maybe we want a log here in case is over the limit ?

Reply:
I'm removing the method, since we now know the bitmap is already downscaled under the size limit, and compressing to webp can only make it smaller. We could also move the size limit method to later in order to test the serialized string for size, however, base64 adds a constant 30% to the bitmap size so we can already predict it and handle it by making the scaled bitmap size at least 30% smaller than the limit.